src/hooks/useCart/useCart.test.tsx

```diff
@@ -182,152 +213,20 @@ describe("useCart", () => {
     jest.resetAllMocks();
   });
 
-  describe("fetch quota on initialisation", () => {
-    it("should initialise the cart with the correct values", async () => {
-      expect.assertions(3);
-      mockGetQuota.mockReturnValueOnce(mockQuotaResSingleIdWithIdentifiers);
-
-      const ids = ["ID1"];
-      const { result, waitForNextUpdate } = renderHook(
-        () => useCart(ids, key, endpoint),
-        { wrapper }
-      );
-      expect(result.current.cartState).toBe("FETCHING_QUOTA");
-
-      await waitForNextUpdate();
-      expect(result.current.cartState).toBe("DEFAULT");
-      expect(result.current.cart).toStrictEqual([
-        {
-          category: "toilet-paper",
-          descriptionAlert: undefined,
-          identifierInputs: [
-            {
-              label: "first",
-              value: "first identifier",
-              textInputType: "STRING",
-              scanButtonType: "BARCODE"
-            },
-            {
-              label: "last",
-              value: "last identifier",
-              textInputType: "STRING",
-              scanButtonType: "BARCODE"
-            }
-          ],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 1,
-          quantity: 1
-        },
-        {
-          category: "chocolate",
-          descriptionAlert: undefined,
-          identifierInputs: [],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 15,
-          quantity: 0
-        }
-      ]);
-    });
-
-    it("should have cart state be NO_QUOTA when no quota is available", async () => {
-      expect.assertions(3);
-      mockGetQuota.mockReturnValueOnce(mockQuotaResSingleIdNoQuota);
-
-      const ids = ["ID1"];
-      const { result, waitForNextUpdate } = renderHook(
-        () => useCart(ids, key, endpoint),
-        { wrapper }
-      );
-      expect(result.current.cartState).toBe("FETCHING_QUOTA");
-
-      await waitForNextUpdate();
-      expect(result.current.cartState).toBe("NO_QUOTA");
-      expect(result.current.cart).toStrictEqual([
-        {
-          category: "toilet-paper",
-          descriptionAlert: undefined,
-          identifierInputs: [],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 0,
-          quantity: 0
-        },
-        {
-          category: "chocolate",
-          descriptionAlert: undefined,
-          identifierInputs: [],
-          lastTransactionTime: transactionTime,
-          maxQuantity: 0,
-          quantity: 0
-        }
-      ]);
-    });
-    it("should have cart state be NO_QUOTA when quota received is invalid", async () => {
```

Comment:
I noticed we removed this test case, and also in useQuota hook we could potentially return negative values for quantities, can we make sure the cart will always receive valid (i.e. non-negative quantities) values?

Reply:
Our back-end actually does not return negative values since there are checks in place, so this test is redundant. Hence, I'm not sure if there's a possibility of the cart receiving negative quantities.

We might have to revisit if there is even a possibility of an invalid quota existing (i.e. negative quantity).

Reply:
yea granted the backend right now does not return invalid values, an accidental mishap could mean the frontend showing strange states.

This is also why we used io-ts, to make sure the backend data isnt malformed to a certain extent. The additional layer of checking of negative values isn't covered by that validation right after the fetch though.

Reply:
I have added another check in `useCart` such that if `cartQuota` contains negative quantities, an error is thrown, and the cart is not set.